src/lib/coda_base/user_command_payload.ml

```diff
@@ -131,36 +132,55 @@ module Body = struct
   type t = Stable.Latest.t =
     | Payment of Payment_payload.Stable.V1.t
     | Stake_delegation of Stake_delegation.Stable.V1.t
+    | Chain_voting of Chain_voting.Stable.V1.t
   [@@deriving eq, sexp, hash, yojson]
 
   let max_variant_size =
     List.reduce_exn ~f:Int.max
-      [Payment_payload.length_in_triples; Stake_delegation.length_in_triples]
+      [ Payment_payload.length_in_triples
+      ; Stake_delegation.length_in_triples
+      ; Chain_voting.length_in_triples ]
 
   module Tag = Transaction_union_tag
 
   let fold = function
     | Payment p ->
-        Fold.(Tag.fold Payment +> Payment_payload.fold p)
+        Fold.(
+          Tag.fold Payment +> Payment_payload.fold p
+          +> Fold.init (max_variant_size - Payment_payload.length_in_triples)
```

Comment:
Was this not here before because `max_variant_size = Payment_payload.length_in_triples`?